ts/components/CiePinpad/index.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  Dimensions,
+  StyleSheet,
+  TextInput,
+  TextInputKeyPressEventData
+} from "react-native";
+
+import variables from "../../theme/variables";
+import { Baseline } from "../Pinpad/Placeholders";
+
+type Props = {
+  description: string;
+  pinLength: number;
+  onPinChanged: (pin: string) => void;
+};
+
+type State = {
+  pin: ReadonlyArray<string>;
+  pinSelected: number;
+  codeSplit: ReadonlyArray<string>;
+};
+
+const styles = StyleSheet.create({
+  placeholderContainer: {
+    flexDirection: "row",
+    justifyContent: "center",
+    backgroundColor: variables.colorWhite
+  },
+  text: {
+    alignSelf: "center",
+    justifyContent: "center",
+    color: variables.colorWhite
+  },
+  placeHolderStyle: {
+    height: 4,
+    marginLeft: 2,
+    marginRight: 2,
+    marginTop: 2
+  }
+});
+
+const CIE_PIN_LENGHT = 8;
+/**
+ * A customized CodeInput component.
+ */
+class CiePinpad extends React.PureComponent<Props, State> {
+  private inputs: ReadonlyArray<TextInput>;
+
+  constructor(props: Props) {
+    super(props);
+    this.inputBoxGenerator = this.inputBoxGenerator.bind(this);
+    this.inputs = [];
+    this.state = {
+      pin: new Array(this.props.pinLength).fill(""),
+      pinSelected: 0,
+      codeSplit: []
+    };
+  }
+
+  private handleOnChangeText = (text: string, index: number) => {
+    const pin = this.updatePin(text, index);
+    this.props.onPinChanged(pin.join(""));
+    // if it is not the last element, change focus on next element
+    // handleOnKeyPress is used to handle the Backspace press
+    if (index + 1 < this.inputs.length) {
+      this.inputs[index + 1].focus();
+      return;
+    }
+  };
+
+  // tslint:disable-next-line: readonly-array
+  private updatePin = (text: string, index: number): string[] => {
+    // tslint:disable-next-line: readonly-array
+    const tempPin = [...this.state.pin];
+    tempPin.splice(index, 1, text);
+    const pin: ReadonlyArray<string> = tempPin;
+    this.setState({ pin });
+    return tempPin;
+  };
+
+  private handleOnKeyPress = (
+    nativeEvent: TextInputKeyPressEventData,
+    index: number
+  ) => {
+    if (nativeEvent.key === "Backspace") {
+      // if it is the first element, do nothing
+      if (index === 0) {
+        return;
+      }
+      // if (index !== 0) {
+      //   this.inputs[index - 1].focus(); // it change the focus on the previous input
+      //   // if (this.props.onDeleteLastDigit) {
+      //   //   this.props.onDeleteLastDigit();
+      //   // }
+      // }
+      // check if a deletion is going.
+
+      if (!this.state.pin[index] || this.state.pin[index].length === 0) {
+        this.updatePin("", index - 1);
+        this.inputs[index - 1].focus();
+      }
+      return;
+    }
+  };
+
+  private inputBoxGenerator = (i: number) => {
+    const width = 36;
+    const margin = 2;
+    const totalMargins = margin * 2 * (this.props.pinLength - 1);
+    const screenWidth = Dimensions.get("window").width;
```

Comment:
Some of they could be constant and placed outside of the class

Reply:
Yes, I had already done it locally :)